### PR TITLE
Add SZip support using libaec

### LINKED ISSUE #120: Add szip support
Issue: szip is a compression module available in the HDF5 distribution . Please enable it

Enabling it should be fairly straightforward if using cmake by adding this flow
-D HDF5_ENABLE_S_LIB_SUPPORT:BOOL=ON

### PR DESCRIPTION
**Pull request by adamcpovey**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Closes #120 

I've only tested the Linux build as that's the machine I have access to.

<br>

(Edit by @jakirkham): More context on SZip support in `libaec` can be found in [this README]( https://gitlab.dkrz.de/k202009/libaec/-/blob/9ea83102b294ee6a93978c6f62db5cd88ed79410/README.SZIP ).

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Very nice. Only osx cross compilation failing.

File: recipe/install.bat [hunk 1]
```diff
@@ -17,6 +17,8 @@ cmake -G "Ninja" ^
       -D HDF5_BUILD_HL_LIB:BOOL=ON ^
       -D HDF5_BUILD_TOOLS:BOOL=ON ^
       -D HDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON ^
+      -D SZIP_LIBRARY:FILEPATH=%LIBRARY_PREFIX%\libsz.so ^
```
Comment (hmaarrfk) [inline recipe/install.bat:20]: Is this line correct. so files are not for windows. Maybe you mean .lib

File: recipe/build.sh [hunk 2]
```diff
@@ -85,6 +85,7 @@ fi
             --host="${HOST}" \
             --build="${BUILD}" \
             --with-zlib="${PREFIX}" \
+            --with-szlib="${PREFIX}" \
```
Comment (hmaarrfk) [inline recipe/build.sh:88]: You might need to export something to help hdf5 identify the configuration for cross compilation

File: recipe/build.sh [hunk 3]
```diff
@@ -96,7 +97,8 @@ fi
             --enable-using-memchecker \
             --enable-static=yes \
             --enable-ros3-vfd \
-	    ${hdf5_disable_tests}
+	    ${hdf5_disable_tests} \
+	    || (cat config.log; false)
```
Comment (jakirkham) [inline recipe/build.sh:101]: This should help us see any details related to configuration errors (atm ones on macOS, but is generally useful outside of this need)

**Comment by jakirkham** [comment]
Added a note in the OP for those interested in knowing how `libaec` provides SZip support

Hope that is ok 🙂

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

File: recipe/install.bat (hunk 1 above)
Comment (adamcpovey) [reply recipe/install.bat:20]: Sorry, I have no idea, this was my best guess as a person that doesn't have access to Windows. (I also don't know if `\lib` is necessary.)

Comment (hmaarrfk) [reply recipe/install.bat:20]: Try to specify with \lib

File: recipe/build.sh (hunk 2 above)
Comment (adamcpovey) [reply recipe/build.sh:89]: Took me a bit to work out what you meant, but I eventually tracked down which bit of the configure was failing. I can't check it here so let's push and see what happens

File: recipe/install.bat [hunk 4]
```diff
@@ -17,7 +17,7 @@ cmake -G "Ninja" ^
       -D HDF5_BUILD_HL_LIB:BOOL=ON ^
       -D HDF5_BUILD_TOOLS:BOOL=ON ^
       -D HDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON ^
-      -D SZIP_LIBRARY:FILEPATH=%LIBRARY_PREFIX%\libsz.so ^
+      -D SZIP_LIBRARY:FILEPATH=%LIBRARY_PREFIX%\libsz.lib ^
```
Comment (hmaarrfk) [inline recipe/install.bat:20]:
```suggestion
      -D SZIP_LIBRARY:FILEPATH=%LIBRARY_PREFIX%\lib\libsz.lib ^
```

**Comment by hmaarrfk** [comment]
Conda is still not detecting libaec being linked to in the windows builds. We need to investigate a little more.

**Comment by adamcpovey** [comment]
> Conda is still not detecting libaec being linked to in the windows builds. We need to investigate a little more.

It just occurred to me to check that repo. Doesn't this line mean they're skipping Windows: https://github.com/conda-forge/libaec-feedstock/blob/caeab9823b3bf204a9d443f966cf8e0ff56862c8/recipe/meta.yaml#L20

**Comment by hmaarrfk** [comment]
No. That is only for windows and visual studio 2008

**Comment by ocefpaf** [comment]
Folks, what is the status of this PR? I was going to tackle this today but found this one open.

**Comment by hmaarrfk** [comment]
I believe this is still true.

> Conda is still not detecting libaec being linked to in the windows builds. We need to investigate a little more.

**Comment by ocefpaf** [comment]
> I believe this is still true.
> 
> > Conda is still not detecting libaec being linked to in the windows builds. We need to investigate a little more.

Thanks Mark! Would folks be OK with a not win version of this? We can open an issue an work on it later.

**Comment by hmaarrfk** [comment]
beyond licensing concerns, which seem to be addressed. I'm not super concerned.

**Comment by jakirkham** [comment]
Yeah we can always raise an issue about Windows support for libaec

Idk if there is a good test we could add to exercise libaec support. That might be worth doing

Having this improvement (even in a limited form) would be an improvement

**Comment by ocefpaf** [comment]
> Yeah we can always raise an issue about Windows support for libaec

+1

> Idk if there is a good test we could add to exercise libaec support. That might be worth doing

@adamcpovey is there something we can add as a test? Maybe a small compress file and try to read it?

I'm keen to merge this soon. Should we remove the changes on Windows and raise the issue? Or leave them there to help the next person who picks this up?

**Comment by jakirkham** [comment]
TBH the test isn't a blocker, but it would just help make sure things continue to work

Perhaps we could repurpose issue ( https://github.com/conda-forge/hdf5-feedstock/issues/120 ) and rescope to Windows only? Though a new issue would work (if preferred)

Would lean towards dropping the Windows changes just to avoid giving the impression this is working. Though no strong feelings if others have preferences

Generally +1 on moving to getting this in ASAP

**Comment by hmaarrfk** [comment]
@ocefpaf did you want to merge this?

**Comment by ocefpaf** [comment]
> @ocefpaf did you want to merge this?

I do! Sorry I forgot about this one. We are missing Windows support right?

**Pull request merged**